gix/src/clone/checkout.rs

```diff
@@ -60,6 +61,54 @@ pub mod main_worktree {
 
     /// Modification
     impl PrepareCheckout {
+        /// Set the `name` of the reference to check out, instead of the remote `HEAD`.
+        /// If `None`, the `HEAD` will be used, which is the default.
+        ///
+        /// Note that `name` should be a partial name like `main` or `feat/one`, but can be a full ref name.
+        /// If a branch on the remote matches, it will automatically be retrieved even without a refspec.
+        pub fn with_ref_name<'a, Name, E>(self, ref_name: Option<Name>) -> Result<PrepareCheckout, Error>
```

Comment:
These three methods can be a single one if a `revspec` is used, see [rev_parse()](https://docs.rs/gix/latest/gix/struct.Repository.html#method.rev_parse) for details.